stl/inc/algorithm

```diff
@@ -10190,15 +10190,22 @@ _NODISCARD constexpr const _Ty& clamp(const _Ty& _Val, const _Ty& _Min_val, cons
     }
 #endif // _ITERATOR_DEBUG_LEVEL == 2
 
-    if (_DEBUG_LT_PRED(_Pred, _Max_val, _Val)) {
-        return _Max_val;
-    }
+    if constexpr (conjunction_v<is_scalar<_Ty>, disjunction<is_same<_Pr, less<>>, is_same<_Pr, less<_Ty>>>>) {

```

Comment:
What about enums for which `operator<` can be overloaded?

Reply:
_Personally_ I'd add this optimization for all basic primitive types for which an overloaded operator is unlikely.
As such I believe specializing this for enums is fine as well. After all this code isn't any less correct, it's just less perfect for _really_ complex `operator<`.

Reply:
I agree that we can let enums with `operator<` overloaded in. As the combination "expensive `operator<` for enum" + "clamp performance for these enums matters" looks extremely unlikely.
I admit this is observable, but the Standard allows making two comparisons in every case.

Reply:
However, I've changed direct `<` usage to `_DEBUG_LT_PRED`, as this enum  `operator <` may be misbehaving (return `true` for both `a < b` and `b < a`), and we have been validation against this in `_ITERATOR_DEBUG_LEVEL == 2`